File: src/main/java/game/GameObject.java
package game;

class GameObject implements Interactive, Drawable
{
    int posX, posY;

    GameObject(int x, int y)
    {
        posX = x;
        posY = y;
    }

    public void interact()
    {
        System.out.println("Interacting with GameObject");
    }

    public void draw()
    {
        System.out.println("Drawing GameObject at position: " + posX + ", " + posY);
    }
}

File: src/main/typescript/game/GameObject.js
"use strict";
exports.__esModule = true;
var GameObject = /** @class */ (function () {
    function GameObject(x, y) {
        this.posX = x;
        this.posY = y;
    }
    GameObject.prototype.interact = function () {
        console.log("Interacting with GameObject");
    };
    GameObject.prototype.draw = function () {
        console.log("Drawing GameObject at position: " + this.posX + " " + this.posY);
    };
    return GameObject;
}());
exports["default"] = GameObject;
